aea/cli/run.py

```diff
@@ -195,35 +197,69 @@ def _profiling_context(period: int) -> Generator:
         sys.stderr = open(os.devnull, "w")
 
 
-def _print_hash_table(ctx: Context, aea: AEA) -> None:
-    """Print hash table of all available components."""
-    hash_data = []
-    ipfs_hash = IPFSHashOnly()
+def _print_instantiated_components(aea: AEA) -> None:
+    """Print table of only components."""
+    components: List[str] = []
     max_col_1_length = 0
-    max_col_2_length = 48
+
     for component_type in [
         ComponentType.PROTOCOL,
         ComponentType.CONNECTION,
         ComponentType.CONTRACT,
         ComponentType.SKILL,
     ]:
-        components = aea.resources.component_registry.fetch_by_type(component_type)
-        for component in components:
-            path = Path(
-                ctx.cwd,
-                VENDOR,
-                component.public_id.author,
-                component_type.to_plural(),
-                component.public_id.name,
-            )
-            if not path.exists():
-                path = Path(
-                    ctx.cwd, component_type.to_plural(), component.public_id.name
-                )
-            hash_data.append(
-                (component.component_id, ipfs_hash.get(str(path), wrap=False))
+        components += [
+            str(component.component_id)
+            for component in aea.resources.component_registry.fetch_by_type(
+                component_type
             )
-            max_col_1_length = max(max_col_1_length, len(str(component.component_id)))
+        ]
+
+    max_col_1_length = max(list(map(len, components)))
+    table_width = max_col_1_length + 6
+    row_separator = "=" * table_width
+    padding = " " * 2
+
+    def format_row(col_1: str) -> str:
+        """Format a row."""
+        return (
+            "|"
+            + padding
+            + col_1
+            + " " * (max_col_1_length - len(col_1))
+            + padding
+            + "|"
+        )
+
+    click.echo("Intantiated packages.")
+    click.echo(row_separator)
+    click.echo(format_row("ComponentId"))
+    click.echo(row_separator)
+    for component_id in components:
+        click.echo(format_row(component_id))
+    click.echo(row_separator)
+    click.echo()
+    click.echo()
+
+
+def _print_all_available_packages(ctx: Context) -> None:
+    """Print hashes for all available packages"""
+    ipfs_hash = IPFSHashOnly()
+    max_col_1_length = 0
+    max_col_2_length = 48
+
+    rows = []
+    for component_type, config_file_name in PACKAGE_TYPE_TO_CONFIG_FILE.items():
+        config_files = Path(ctx.cwd).glob(f"**/{config_file_name}")
```

Comment:
I also don't like the glob approach. We know from the framework where exactly the files are. Let's not indiscriminately find any config file in the file tree. What if someone misplaces a file? It will show up...

The previous implementation was much cleaner in this regard (210-226).

Reply:
> This is not good separation of concerns. Firstly, we probably have an object with this list already accessible. 

So I looked over the `AEA` class implementation and didn't find any method or property which has a list of all available packages in a project.

> I also don't like the glob approach. We know from the framework where exactly the files are. Let's not indiscriminately find any config file in the file tree. What if someone misplaces a file? It will show up...

I don't think it's a common occurrence when a component config file is misplaced and even if a misplaced config file shows up it would help detect that some file is misplaced (Think of this as a feature, not a bug 😅). Also I looked at the frameworks way of finding all the available packages in a project, It's similar to what you when you refactored this method. But instead of using initialised components it uses the aea config to iterate over all the public ids and generate path for the package. I can create a utility method that does the same and I think it might be useful somewhere else well 

> Secondly, it should not be the concern of this method to know about where config files are stored and how they are named. You need to extract this or reuse existing functionality (preferred!!!).

I'll create a utility method which will return a list of paths to all available packages in a given directory.